### PR TITLE
Remove duplicate code between web frontend and tests

### PR DESCRIPTION
**Pull request by smarr**
The goal is to finish the transition to TypeScript and remove code duplication where not necessary.

The changes should improve the overall code structure.
We are also going to move more code to a TypeScript style away from plain JavaScript.

### CODE REVIEW

File: tools/kompos/src/messages.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  sourceSection: SourceSection;
+  methodName: string;
+}
+
+interface TopFrame {
+  arguments: string[];
+  slots:     IdMap<string>;
+}
+
+export type Message = SourceMessage | SuspendEventMessage | MessageHistoryMessage;
```
Comment (ctrlpz) [inline tools/kompos/src/messages.ts]: What is the goal of it type of message?

File: tools/kompos/src/visualizations.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ var width = 360,
 
 function hasSelfSends(actorId, messages) {
   for (var i in messages) {
-    if (messages[i].sender == "actorId") {
+    if (messages[i].sender === "actorId") {
       return true;
     }
   }
```
Comment (ctrlpz) [inline tools/kompos/src/visualizations.ts]: What is this function for? Where it is used?

File: tools/kompos/src/messages.ts (hunk 1 above)
Comment (smarr) [inline tools/kompos/src/messages.ts]: Not sure what you are asking for. Do you want documentation? Things are going to change soon (including naming). So, I'd not yet invest the time on that.

File: tools/kompos/src/messages.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  topFrame: TopFrame;
+  id: string;
+}
+
+export interface MessageHistoryMessage {
+  type: "messageHistory";
+  messageHistory: any; // TODO
+}
+
+export type BreakpointData = LineBreakpointData | SendBreakpointData | AsyncMethodRcvBreakpointData;
```
Comment (smarr) [inline tools/kompos/src/messages.ts]: Stuff is going to change around soon again. So, I put them all in here for the moment.
There is also currently rather bad choice of terms with Message/Respond. That's very depending on what you consider sender/receiver. So, doesn't make sense universally. We'll need to find better terminology there, too.

File: tools/kompos/src/visualizations.ts (hunk 2 above)
Comment (smarr) [inline tools/kompos/src/visualizations.ts]: `visualizations.ts` is for graphical things, the actor history. While `view.ts` is for normal UI elements.

File: tools/kompos/src/messages.ts (hunk 1 above)
Comment (ctrlpz) [inline tools/kompos/src/messages.ts]: ok, but this code in message.ts file was first implemented in basic-protocol.ts, so the idea now is that it is not use only for testing, instead now Kompos uses those interfaces and the rest of your implementation and in basic-protocol.ts is only imported, right?

File: tools/kompos/src/messages.ts (hunk 3 above)
Comment (ctrlpz) [inline tools/kompos/src/messages.ts]: ok

File: tools/kompos/src/visualizations.ts (hunk 2 above)
Comment (ctrlpz) [inline tools/kompos/src/visualizations.ts]: ok thanks

File: tools/kompos/src/messages.ts (hunk 1 above)
Comment (smarr) [inline tools/kompos/src/messages.ts]: Yes, the idea is that there is only on set of type definitions for the tests and the web frontend.

**Pull request merged**